src/main/java/com/google/api/generator/spring/composer/SpringPropertiesClassComposer.java

```diff
@@ -315,6 +322,52 @@ private static MethodDefinition createSetterMethod(
         .build();
   }
 
+  private static MethodDefinition createDurationSetterMethod(
+      TypeNode thisClassType, String propertyName, TypeNode returnType) {
+    Variable propertyVar = Variable.builder().setName(propertyName).setType(returnType).build();
+    Variable argumentVar =
+        Variable.builder()
+            .setName(propertyName)
+            .setType(staticTypes.get("java.time.Duration"))
+            .build();
+    Expr thisExpr = ValueExpr.withValue(ThisObjectValue.withType(thisClassType));
+
+    MethodInvocationExpr durationToStringExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(VariableExpr.withVariable(argumentVar))
+            .setMethodName("toString")
+            .setReturnType(TypeNode.STRING)
+            .build();
+
+    MethodInvocationExpr parsedDurationExpr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(staticTypes.get("org.threeten.bp.Duration"))
+            .setMethodName("parse")
+            .setArguments(durationToStringExpr)
+            .setReturnType(staticTypes.get("org.threeten.bp.Duration"))
+            .build();
+
+    AssignmentExpr propertyVarExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(
+                VariableExpr.withVariable(propertyVar)
+                    .toBuilder()
+                    .setExprReferenceExpr(thisExpr)
+                    .build())
+            .setValueExpr(parsedDurationExpr)
+            .build();
+
+    String methodName = "set" + CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, propertyName);
```

Comment:
Hmm good question - looks like under the hood `JavaStyle.ToUpperCamelCase()` uses `CaseFormat` to [convert from lower snake to upper camel case](https://github.com/googleapis/gapic-generator-java/blob/ee5d7a5985d298e6d96dc700805870f3e362b1de/src/main/java/com/google/api/generator/gapic/utils/JavaStyle.java#L53), whereas we are converting from lower camel in this line.

I'm inclined to keep the current implementation but practically they do yield the same result, since here we just need the first letter of `propertyName` to change from lower to upper case.